packages/extension-commerce-bm-seo/src/types/config.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import type {ApplicationExtensionConfig} from '@salesforce/pwa-kit-extension-sdk/types'
+
+// Defines the map of resource types to component display name.
+// https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-seo?meta=getUrlMapping
+type ResourceTypeToComponentMap = {
+    category?: string
+    product?: string
+    content_asset?: string
```

Comment:
i made these optional so that customers only need to define the resource types that they want to map in their app.
e.g., customer only uses `product` url mappings